ddl_checker/main.go

```diff
@@ -0,0 +1,216 @@
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bufio"
+	"database/sql"
+	"flag"
+	"fmt"
+	_ "github.com/go-sql-driver/mysql"
+	"github.com/pingcap/tidb-tools/pkg/ddl-checker"
+	"os"
+	"strconv"
+	"strings"
+)
+
+var (
+	modCode                                          = 0
+	executableChecker *ddl_checker.ExecutableChecker = nil
+	reader            *bufio.Reader                  = nil
+	db                *sql.DB                        = nil
+
+	host     = flag.String("host", "127.0.0.1", "MySQL host")
+	port     = flag.Int("port", 3306, "MySQL port")
+	username = flag.String("user", "root", "User name")
+	password = flag.String("password", "", "Password")
+	schema   = flag.String("schema", "test", "Schema")
+)
+
+const (
+	WelcomeInfo = "ExecutableChecker: Check if SQL can be successfully executed by TiDB\n" +
+		"Copyright 2018 PingCAP, Inc.\n\n" +
+		"You can switch modes using the `SETMOD` command.\n" +
+		"Auto mode: The program will automatically synchronize the dependent table structure from MYSQL " +
+		"and delete the conflict table\n" +
+		"Query mode: The program will ask you before synchronizing the dependent table structure from MYSQL\n" +
+		"Manual mode: This program does not perform anything other than executing the input SQL.\n" +
+		"SETMOD Usage: SETMOD <MODCODE>; MODCODE = [0(Auto), 1(Query), 2(Manual)]\n"
+)
+
+func main() {
+	initialise()
+	printWelcome()
+	mainLoop()
+	//handler("ALTER TABLE table_name MODIFY column_1 int(11) NOT NULL;")
+	destroy()
+}
+
+func initialise() {
+	var err error
+	reader = bufio.NewReader(os.Stdin)
+	executableChecker, err = ddl_checker.NewExecutableChecker()
+	if err != nil {
+		fmt.Printf("[DDLChecker]Init failed,Can't create ExecutableChecker: %s\n", err.Error())
+		os.Exit(1)
+	}
+	executableChecker.Execute("use test;")
+	db, err = sql.Open("mysql", fmt.Sprintf("%s:%s@tcp(%s:%d)/%s",
+		*username, *password, *host, *port, *schema))
+	if err != nil {
+		fmt.Printf("[DDLChecker]Init failed,Can't open mysql database: %s\n", err.Error())
+		os.Exit(1)
+	}
+
+}
+
+func destroy() {
+	executableChecker.Close()
+	db.Close()
+}
+
+func mainLoop() {
+	var input string
+	var err error
+	for cont := true; cont; cont = handler(input) {
+		fmt.Printf("[%s] > ", modeName())
+		input, err = reader.ReadString(';')
+		if err != nil {
+			fmt.Printf("[DDLChecker]Read Stdin Error: %s\n", err.Error())
+			os.Exit(1)
+		}
+	}
+}
+func handler(input string) bool {
+	// cmd exit
+	lowerTrimInput := strings.ToLower(strings.TrimSpace(input[:len(input)-1]))
```

Comment:
Use `strings.TrimRight(input, "; \t\n\r")` to removing the trailing semicolon. (or `TrimRightFunc` if you do care about Unicode)